### PR TITLE
[DRAFT] Straightjackets PR, bring'em home

### PR DESCRIPTION
**Pull request by TheOrleans**
Reverts BoHBranch/BoH-Bay#590
IMPORTANT NOTE: Fuzzy cuffs will be kept out, the other changes regarding wrapping people won't be changed. This PR is only about straightjackets. Commits will be updated. 

**Reasons why this PR was made**
Here we go. This is a revert PR, so I'm going to go ahead and explain why this should be done.
- Let's rembember how the vote went. I know the PR was updated after that, but it never got enough support to have them removed
[image](https://user-images.githubusercontent.com/4706052/78464679-1c9bb480-76a1-11ea-9530-b223f0fd98a9.png)
[image](https://i.gyazo.com/849c9f3016111a49955ff810939a0d64.png)
This PR never got decent support at the moment, now let's see how the situation is right now.
I talked about it on Discord, and you know? I got support, at least more upvotes than downvotes, which is not something that the removing PR can say
[image](https://i.gyazo.com/6debd73053e31067067c9cbb093973b5.png)
I know the downsides of straightjackets. Security abusing them, putting someone in them and forgetting about that people for the rest of the round. Medical giving them away like candy... but, is removal instead of education the right way to adress this?

We need straightjackets to stop people from acting like psychos. That kind of people who get out of Brig by smashing a window with any object 100 times until they break, and then shock themselves by smashing the powered grille. They need to have something that tells them "if you behave like a psycho or a madman, expect to be treated like one"

Of course, people who escape without the psycho behaviour shouldn't be put in them. Smart or technical escaping (C4, disassembling, or whatever). And we need to make sure EVERYONE understands this. 

If Medical gives them away without actually sending a doctor to inspect the subject, bwoink.
If Security abuses them, bwoink.

Removal is not the solution, education is.

**1st PR Update after discussion on Discord**

According to some suggestions on Discord, I am inclined to make STRAIGHTJACKETS something you can escape. The idea is to make them have a **7 minutes timer + Shaking animation** so it's quite obvious. @Roland410 @Superhats @SouthyOsmanli what do you think

**2nd PR Update after poll on Discord**

I think it's clear. Community wants straightjackets back, and they want them escapable, so that's what I am doing.

I am making this PR a Draft until it's finished.
The current idea is:
- 4 or 5 minutes resist timer
- Shaking effect when resisting.

### CODE REVIEW

**Comment by Roland410** [comment]
Straightjackets are literal cancer, and the only reason it wasn't purged from code is because @Burger said he would make them escapeable, so we agreed on only removing it from the map and other spawns (such as the random spawner).
They were, and if brought back WILL be abused by security again to cuck antags then leave them to rot.
It doesn't matter if medical, science, engineering, supply or any other department has them in their storage, because they WILL give it over to security whenever asked to do so, because why not.
It being unescapeable is the real problem here, and bringing it back in their state will just make more work for staff, who, if the antag doesn't straight up just ghost (wouldn't be the first time it happened) now have to deal with more ahelps because of this, have to investigate if the 'jacketing' is legit or not, and if they decide it wasn't (it usually wasn't), then it also breaks the flow of the round, as now you have no IC explanation why you are taking them out of the jacket.
Even in situations where it was "justified" (mostly runecult), it just allowed security to ignore any roleplay and go hunt more valids, especially considering there were SIX of the jackets (along with blindfolds and muzzles) they could take from medical.
This means in cult round where they didn't manage to get too many converts done yet, security could jacket them all once they were found because "crazy man cuts hand".

All in all, the existence of straightjackets is a mistake that should be forgotten in the dust.

**Comment by SouthyOsmanli** [comment]
The PR was pushed through by Burger in the first place for balance reasons since people misused them heavily and antags also did. They are unbreakable to get out of, meaning the player has no way of struggling out as they would cuffs. You are forever trapped until someone else takes them off of you or unless you /somehow/ get psionic telekensises to remove them as you can cuffs. 

This created issues where players were shoved in jackets for small issues and left buckled to beds for at least an hour; either being forgotten or so forth. If you have to use one it's usually due to an issue that is not only IC but typically OOC shittery; which needs to be A-helped and dealt with anyhow.

**Comment by TheOrleans** [comment]
@Roland410 Straightjackets SHOULDN'T be escapeable. They are not body-size handcuffs, they are STRAIGHTJACKETS. If you are put in one, you deserve to be in it, and if you aren't ahelp it. A lot of stuff causes problems and ahelps, and that argument is quite weak. Educate the people on how to use the straightjackets properly, stablish a procedure where medical has to actually go and inspect the person, don't take the straightjackets away.

@SouthyOsmanli ... People who leave prisoners inside a cell, buckled to a bed, in a straightjacket without a PROPER reason should be educated or punished, taking straightjackets away is not a decent solution (imo)

**Comment by Roland410** [comment]
If you think "educating" people solves issues, then you may want to apply to be a game admin or something. From a code pespective and basing it on previous experiences, it doesn't work.
People have abused it in the past, and no radical change happened that would prevent them from being abused if they were to be brought back.

I agree with Southy mostly, although I would highlight the part in my comment above, that if we really stretch it, it could apply for cultists (the proper use part), but even then it means you're effectively blocking the round from progressing.

**Comment by SouthyOsmanli** [comment]
@TheOrleans I see **no** reason to re-add them. They did not encourage RP, any people commonly put into the jackets were banned for LRP behavior as is as was their purpose and they add nothing to the game itself besides an uncounterable restraint.

Your issue you're stating for the re-adding on them is quite literally an admin issue. If someone is being a massive LRP shitter, you A-help it anyway. The usage of them has no purpose. They were rarely used for that reason during their addition as someone who commonly saw them used in Brig. They were just used as an alternative to executing someone and just leaving him buckle-cuffed for being annoying to Security.

Psychos always got handled by an admin upon reporting the horrid behavior and Sec that used them otherwise always got handled. Their addition was pointless besides the very rare reasonable insane/suicide case - in which admins spawn the jacket for upon being asked if it is within reason. 

A prime example of a decent situation for their use was when a guy was mindslaved to kill himself so medical needed to restrain him due to his valid 'psycho' like activity. Most bad examples is just buckle cuffing a prisoner in perma because be broke something any annoyed you.

**Comment by BurgerLUA** [comment]
This needs a poll and more discussion.

**Comment by TheOrleans** [comment]
I keep thinking that straightjackets let you handle people IC, apart from the possible OOC punishing, and that, if you need a process to use them:
- Approval from the Mentalist or if not present highest ranking medical person
- Approval from the Brig Chief or if not present highest ranking sec person

And if everyone understands when to use them, they could be good.

Finally I would totally like a poll on Discord, to see what the people think about it, since previous polls were affected by the fuzzy cuffs and that stuff. A poll just about the straightjackets

File: code/game/objects/items/weapons/handcuffs.dm [hunk 1]
```diff
@@ -180,6 +180,15 @@ var/last_chew = 0
 	cuff_type = "duct tape"
 	health = 50
 
+/////////
+// Fuzzy Cuffs. Kill me.
+/////////
+
+/obj/item/weapon/handcuffs/fuzzy
```
Comment (Superhats) [inline code/game/objects/items/weapons/handcuffs.dm:187]: That wasn't mentioned in your PR.

File: code/game/objects/random/random.dm [hunk 2]
```diff
@@ -1036,6 +1036,7 @@ something, make sure it's not in one of the other lists.*/
 				/obj/item/clothing/glasses/welding/superior = 7,
 				/obj/item/clothing/head/collectable/petehat = 4,
 				/obj/item/clothing/suit/storage/vest/merc = 3,
+				/obj/item/clothing/suit/straight_jacket = 6,
```
Comment (Superhats) [inline code/game/objects/random/random.dm:1039]: I don't know if adding straight jackets as maint loot is a good idea.

**Comment by Superhats** [comment]
Really does need a poll.

File: code/game/objects/items/weapons/handcuffs.dm (hunk 1 above)
Comment (Superhats) [reply code/game/objects/items/weapons/handcuffs.dm:187]: Nevermind, I am blind.

Comment (TheOrleans) [reply code/game/objects/items/weapons/handcuffs.dm:187]: As I said in the second line of the PR (literally, second line), that is in commits cause I created the PR as a revert to another PR which also removed the Fuzzy Cuffs but I will only add straightjackets

If the poll is supportive towards the idea, I will change the commits to only add the straighjackets, the rest will remain as it is. Fuzzy cuffs are not allowed by Carl anyways, so it's a no-go anyways. 

Conversation resolved, fuzzy cuffs won't be added back.

File: code/game/objects/random/random.dm (hunk 2 above)
Comment (TheOrleans) [reply code/game/objects/random/random.dm:1039]: Uhm, I didn't think about this, but I can agree. They will only be added to the Medical Storage.

Conversation resolved, straightjackets WON'T be maint loot.

**Comment by TheOrleans** [comment]
> I'd rather go for a simple re-mapping them instead of a full revert of my PR as it touches other stuff such as the fuzzy cuffs, also really needs a poll.

If the poll is supportive towards straightjackets coming back, I will update the commits to only re-add the straightjackets to the medical storage and psychiatrist closet, I made the PR to get the community to vote on straightjackets.

No problem, just make sure to NOT MERGE IT until I update it! :D

**Comment by Superhats** [comment]
DNM as requested.

**Comment by Roland410** [comment]
7 minutes is ridiculous when all someone has to do is move you away one tile.

**Comment by TheOrleans** [comment]
> 7 minutes is ridiculous when all someone has to do is move you away one tile.

6 maybe, cuffs are 2

**Comment by Sparrowbird611** [comment]
I am EXCEPTIONALLY late to this show, but I'm gonna say what I think now.

I'm against returning these, but if you are, you should A: leave the idiotic shaking out and B: Make them subdue for one-two minutes maximum. These stack with handcuffs and chair breaking, adding up to a total of five minutes of imprisonment time (This is if I'm remembering correctly, and breaking out of a chair is only one minute.) that give off incredibly obvious messages and can be stopped so easily a brain dead corpse could do it.

All in all, there's no real reason to add these and are actively detrimental as they screw with numerous mechanics (Freedom implants being the most obvious in my mind) and your current written plan is horrific. If you're dead set on returning them, don't do it like that.

**Comment by SilvernKell** [comment]
I'm with sparrow mostly
Honestly I think they should just be like an alternate to cuffs with a SLIGHTLY longer escape time and the shaking, and they should be rare like only 1 for medical. It has uses because they can affect people with only 1 hand and also sometimes it's just easier to put a shitter who keeps saying "let me die" in the straitjacket whilst waiting for an admin or if there isn't an admin on, and because it makes sense if you are trying to take them to a counselor.
Or sometimes you get people who have just snapped, from touching anomalies like the pool of life/pillars on other worlds, in which case they would ONLY be jacketed whilst science is talking to them + trying to find a way to restore them or calm them and they shouldn't get bored.
You really don't need more then like a single one, and security shouldn't be using them unless a physician puts it on whilst working with the person, trying to discuss their motive and the issue.

**Pull request closed without merging**